src/pykeen/nn/emb.py

```diff
@@ -34,30 +41,99 @@ def forward(
         """
         raise NotImplementedError
 
+    def get_in_canonical_shape(
+        self,
+        indices: Optional[torch.LongTensor] = None,
+        reshape_dim: Optional[Sequence[int]] = None,
+    ) -> torch.FloatTensor:
+        """Get representations in canonical shape.
+
+        :param indices:
+            The indices. If None, return all embeddings.
+        :param reshape_dim:
+            Optionally reshape the last dimension.
+
+        :return: shape: (batch_size, num_embeddings, d)
+        """
+        x = self(indices=indices)
+        if indices is None:
+            x = x.unsqueeze(dim=0)
+        else:
+            x = x.unsqueeze(dim=1)
+        if reshape_dim is not None:
+            x = x.view(*x.shape[:-1], *reshape_dim)
+        return x
+
     def reset_parameters(self) -> None:
         """Reset the module's parameters."""
 
     def post_parameter_update(self):
         """Apply constraints which should not be included in gradients."""
 
 
+@dataclasses.dataclass
+class EmbeddingSpecification:
+    """An embedding specification."""
+
+    # embedding_dim: int
+    # shape: Optional[Sequence[int]] = None
+
+    initializer: Optional[Initializer] = None
+    initializer_kwargs: Optional[Mapping[str, Any]] = None
+
+    normalizer: Optional[Normalizer] = None
+    normalizer_kwargs: Optional[Mapping[str, Any]] = None
+
+    constrainer: Optional[Constrainer] = None
+    constrainer_kwargs: Optional[Mapping[str, Any]] = None
+
+    regularizer: Optional[Regularizer] = None
+
+    def make(
```

Comment:
What is the advantage of allowing to create an Embedding directly from the Specification class, and not only from the Embedding class?